### PR TITLE
Setup gatsby, basic sitewrap, build system

### PR DESCRIPTION
**Pull request by mattbowen-usds**
This is related to #48 in the v3 planning project. It adds the following capabilities:

- A basic gatsby site that uses the [trussworks USWDS](https://github.com/trussworks/react-uswds).
- Implements a simple header and footer
- Implements a single markdown page type, as an example
- Add "production" and preview build actions that run on pushes to the appropriate branches.

### CODE REVIEW

File: tech-far-hub/src/pages/case-studies/{mdx.frontmatter__slug}.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      {children}
+    </Layout>
+  )
+}
+
+export default CaseStudyPage
+
+export const Head: HeadFC = () => <title>TechFAR Hub</title>
+
+export const query = graphql`
```
Comment (vim-usds) [inline tech-far-hub/src/pages/case-studies/{mdx.frontmatter__slug}.tsx:22]: Loving how easy it is to make GQL calls right in the component!

File: tech-far-hub/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,102 @@
+{
```
Comment (vim-usds) [inline tech-far-hub/tsconfig.json:1]: Is this how the tsconfig came out of the box? With every option available to uncomment? 🤯 

If so, that's so cool!

**Review by vim-usds — APPROVED** [review]
Lightening fast Matt! Amazing work!

File: .github/workflows/deploy-staging.yml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        run: echo "branch=$(echo ${GITHUB_REF#refs/heads/} | sed 's/\//:/g')" >> $GITHUB_ENV
+        id: extract_branch
+      - name: Setup node 18
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
+      - name: Build pages
+        uses: enriikke/gatsby-gh-pages-action@v2
+        env:
+          DEPLOY_TYPE: 'staging'
```
[the remark this answers is not in the document]
Comment (mattbowen-usds) [reply .github/workflows/deploy-staging.yml:25]: Not yet! I think you're right, I should get rid of it until I need it

File: tech-far-hub/tsconfig.json (hunk 2 above)
Comment (mattbowen-usds) [reply tech-far-hub/tsconfig.json:1]: It is! And until I know what I'm doing, I'm leaving it lol!

**Pull request merged**